Signal/src/call/CallService.swift

```diff
@@ -773,6 +779,38 @@ protocol CallServiceObserver: class {
     func setHasLocalVideo(hasLocalVideo: Bool) {
         assertOnSignalingQueue()
 
+        let authStatus = AVCaptureDevice.authorizationStatus(forMediaType:AVMediaTypeVideo)
+        switch authStatus {
+        case .notDetermined:
+            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusNotDetermined")
+            break
+        case .restricted:
+            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusRestricted")
+            break
+        case .denied:
+            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusDenied")
+            break
+        case .authorized:
+            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusAuthorized")
+            break
+        }
+
+        // We don't need to worry about the user granting or remotinv this permission
+        // during a call while the app is in the background, because changing this
+        // permission kills the app.
+        if authStatus != .authorized {
+            DispatchQueue.main.async {
+                let title = NSLocalizedString("CAMERA_PERMISSION_MISSING_TITLE", comment: "Alert title when camera is not authorized")
```

Comment:
This is pretty similar to https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/util/UIViewController+CameraPermissions.m

Is there anything we could reuse or extract? That code has the added nicety of pointing the user to the settings app.